content-review/impl/turnitin-oc/src/main/java/org/sakaiproject/contentreview/turnitin/oc/ContentReviewServiceTurnitinOC.java

```diff
@@ -625,7 +625,18 @@ public void popAdvisor(SecurityAdvisor advisor) {
 		// Send request:
 		int responseCode = connection.getResponseCode();
 		String responseMessage = connection.getResponseMessage();
-		String responseBody = IOUtils.toString(connection.getInputStream(), StandardCharsets.UTF_8);
+		String responseBody;
+		if (responseCode < 200 || responseCode >= 300)
+		{
+			// getInputStream() throws an exception in this case, but getErrorStream() has the information necessary for troubleshooting
+			responseBody = IOUtils.toString(connection.getErrorStream(), StandardCharsets.UTF_8);
+			log.warn("Turnitin response code: " + responseCode + "; message: " + responseMessage + "; body:\n" + responseBody);
+		}
+		else
+		{
+			responseBody = IOUtils.toString(connection.getInputStream(), StandardCharsets.UTF_8);
+			log.debug("Turnitin response code: " + responseCode + "; message: " + responseMessage + "; body:\n" + responseBody);
```

Comment:
to avoid unnecessarily converting the InputStream to a string, you can use new-style of logging:

log.debug("Turnitin response code: {} ..........", variable);

Reply:
I'll keep that in mind. In this case responseBody is used in the code below